File: src/3226.number-of-bit-changes-to-make-two-integers-equal.3508/solution.ts
function minChanges (n: number, k: number): number {
    if (k > n) {
        return -1;
    }
    let cnt = 0;
    while (n > 0) {
        const a = n & 1;
        const b = k & 1;
        if (a !== b) {
            if (a === 1) {
                cnt++;
            } else {
                return -1;
            }
        }
        n >>= 1;
        k >>= 1;
    }
    return cnt;
}

File: src/3226.number-of-bit-changes-to-make-two-integers-equal.3508/solution.java
class Solution {
    public int minChanges(int n, int k) {
        int result = 0;
        while (n != 0){
            int digiN = n&1;
            int digiK = k&1;
            if(digiN == 0 && digiK == 1){
                return -1;
            }
            if(digiN == 1 && digiK == 0){
                result++;
            }

            n >>= 1;
            k >>= 1;
        }
        return k != 0? -1:result;
    }
}